Reduce page increment of GTK wxSpinButton for narrow range

RichardReznicek opened the pull request:
Gradually reduce the page increment in the case of a narrow range for
convenience and to limit possible `EVT_SPIN_UP`/`EVT_SPIN_DOWN` ambiguity
when wrapping is on. 

This covers the issue of sending wrong events 

> when the arrows are held pressed for long enough 

mentioned in https://github.com/wxWidgets/wxWidgets/pull/1764

However, transitions from minimum to maximum or vice versa by secondary mouse button clicks still cannot be prevented from producing wrong events if wrapping is allowed.

RichardReznicek committed:
Reduce page increment of GTK wxSpinButton for narrow range

Gradually reduce the page increment in the case of a narrow range for
convenience and to limit possible EVT_SPIN_UP/EVT_SPIN_DOWN ambiguity
when wrapping is on.

vadz commented on src/gtk/spinbutt.cpp:
```diff
@@ -186,6 +186,16 @@ void wxSpinButton::SetRange(int minVal, int maxVal)
+    // (The maximal page increment of 10 is consistent with the default page increment set
+    // by calling gtk_spin_button_new_with_range(0, 100, 1) in wxSpinButton::Create().)
+    const int rng = maxVal - minVal;
+    const int inc = 10 - ( (rng < 50) ? 5 : 0 ) - ( (rng < 20) ? 3 : 0 ) - ( (rng < 10) ? 1 : 0 );
```
Thanks, I don't disagree with the idea (although I'm not sure if people are not going to be surprised by this), but IMO this would be much more clear if it were written as a sequence of `if`s or, if you really want to keep it in the same expression as,
```suggestion
    const int inc = range < 10 ? 1 : range < 20 ? 2 : range < 50 ? 5 : 10;
```
(I've also renamed the variable to avoid any impression that the increment is determined using the Random Number Generator).

RichardReznicek replied:
OK, I have rewritten it using a set of `if` statements.

RichardReznicek committed:
Rewrite page increment selection using if statements

The selection of page increment in GTK wxSpinButton:SetRange() was
rewritten from a single line to a set of if statements for better
readibility.

vadz commented:
Thanks, but I just don't understand the advantage of going down (50, 20, 10) rather than checking the range directly which seems much more natural, so I've reversed the order of checks. Hopefully you don't mind, but please let me know if you do.
